stdlib/sys/socket.gr

```diff
@@ -0,0 +1,422 @@
+/**
+ * @module Socket: Utilities for handling server sockets
+ * 
+ * @example import Socket from "sys/socket"
+ */
+
+import WasmI32, {
+  add as (+),
+  mul as (*),
+  shl as (<<),
+  shrS as (>>),
+  ne as (!=),
+  or as (|),
+} from "runtime/unsafe/wasmi32"
+import File from "sys/file"
+import Wasi from "runtime/wasi"
+import Memory from "runtime/unsafe/memory"
+import { allocateString, tagSimpleNumber } from "runtime/dataStructures"
+import String from "string"
+import Bytes from "bytes"
+
+
+/**
+ * Own file descriptor enum
+ *
+ * For not using the same as in the file api
+ *
+ * @since v0.5.14
+ */
+export enum FileDescriptor {
+  FileDescriptor(Number),
+}
+
+/**
+ * Socket fd flags
+ *
+ * All fd_flags specific for the socket handling
+ *
+ * @since v0.5.14
+ */
+export enum SockFlags {
+  Append,
+  DSync,
+  NonBlock,
+  RSync,
+  Sync,
+}
+
+/**
+ * Socket send input flags
+ *
+ * All si_flags for socket handling
+ *
+ * @since v0.5.14
+ */
+export enum SiFlags {
```

Comment:
It's a bit more verbose, but I'd advocate labeling this and the next types as `SendInputFlags`, `ReadInputFlags`, etc.